packages/connection/src/Manager.php

```diff
@@ -124,7 +137,281 @@ public function verify_signature() {
 	 * @return Integer zero on success, or a bitmask on failure.
 	 */
 	public function register() {
-		return 0;
+		// TODO: Tracking can't yet function without static Jetpack methods.
+
+		add_action( 'pre_update_jetpack_option_register', array( '\Jetpack_Options', 'delete_option' ) );
+		$secrets = $this->generate_secrets( 'register', get_current_user_id(), 600 );
+
+		if (
+			empty( $secrets['secret_1'] ) ||
+			empty( $secrets['secret_2'] ) ||
+			empty( $secrets['exp'] )
+		) {
+			return new \WP_Error( 'missing_secrets' );
+		}
+
+		// better to try (and fail) to set a higher timeout than this system
+		// supports than to have register fail for more users than it should
+		$timeout = $this->set_min_time_limit( 60 ) / 2;
+
+		$gmt_offset = get_option( 'gmt_offset' );
+		if ( ! $gmt_offset ) {
+			$gmt_offset = 0;
+		}
+
+		$stats_options = get_option( 'stats_options' );
+		$stats_id      = isset( $stats_options['blog_id'] )
+				  ? $stats_options['blog_id']
+				  : null;
+
+		$args = array(
+			'method'  => 'POST',
+			'body'    => array(
+				'siteurl'         => site_url(),
+				'home'            => home_url(),
+				'gmt_offset'      => $gmt_offset,
+				'timezone_string' => (string) get_option( 'timezone_string' ),
+				'site_name'       => (string) get_option( 'blogname' ),
+				'secret_1'        => $secrets['secret_1'],
+				'secret_2'        => $secrets['secret_2'],
+				'site_lang'       => get_locale(),
+				'timeout'         => $timeout,
+				'stats_id'        => $stats_id,
+				'state'           => get_current_user_id(),
+				'site_created'    => $this->get_assumed_site_creation_date(),
+				'jetpack_version' => JETPACK__VERSION,
```

Comment:
Should we use the constants package for this one?

Reply:
Yes, I'm hoping to merge Michael's #12717 into this PR once it gets merged.

Reply:
Fixed that, thank you!